### ISSUE TITLE
Dependencies not included in package

### ISSUE THREAD

**Issue by ss2165** [CQCL/tket2#476]
Originally reported by @sjdilkes 
The wheels being built are not including dependencies. This is probably because the dependencies are only under the poetry tool section of `pyproject.toml`. They may need to be duplicated. Is there a way that avoids duplication?

**Comment by aborgna-q** [comment]
Short resume of various issues:
- [PEP621](https://peps.python.org/pep-0621/) specifies that dependencies should go under `[project]`
- Maturin does not support poetry's custom pyproject fields. It expects it to follow PEP621.
  [maturin#632](https://www.github.com/PyO3/maturin/issues/632)
- Poetry has had an open issue about it for almost 4 years:
  https://github.com/python-poetry/poetry/issues/3332
- There's a recent draft PR in poetry that implements that. It will be included in "poetry 2.0.0", once that comes out.
  [poetry#9135](https://www.github.com/python-poetry/poetry/pull/9135)
  
So in short; we'll have to duplicate things for now :/
